### ISSUE TITLE
[BUG] Series.replace returning null values after label_encoding

### ISSUE THREAD

**Issue by rjzamora** [rapidsai/cudf#3785]
**Describe the bug**
I am seeing strange output for the `replace` method when the underlying `Series` object is generated by the `label_encoding` method.  More specifically, `replace` is erroneously turning most of the elements into `null` values (see reproducer below).

**Steps/Code to reproduce bug**

```python
import cudf
import numpy as np

_cats = ["cat", "dog", "bird"]
series = cudf.Series(np.random.choice(_cats, 100))
encoded = series.label_encoding(_cats)
encoded.replace(0, 3)
```

Output:
```
0        2
1        3
2     null
3     null
4     null
      ... 
95    null
96    null
97       3
98    null
99    null
Length: 100, dtype: int32
```

Note that if I first take a slice of `encoded`, the output is correct:
```
0     3
1     1
2     3
3     1
4     1
     ..
95    1
96    1
97    1
98    2
99    1
Length: 100, dtype: int32
```

**Expected behavior**
I would not expect the `replace` method to create new null values.

**Environment overview (please complete the following information)**
 - Environment location: Bare-metal
 - Method of cuDF install: from source

**Environment details**
<details><summary>Click here to see environment details</summary><pre>
     
     **git***
     commit 3d2680253f6193f9734539cd477c341c5da468e5 (HEAD, jperez999/master)
     Author: Julio Perez <juperez999@hotmail.com>
     Date:   Wed Jan 15 00:58:50 2020 +0000
     
     all except dask green
     **git submodules***
     
     ***OS Information***
     DGX_NAME="DGX Server"
     DGX_PRETTY_NAME="NVIDIA DGX Server"
     DGX_SWBUILD_DATE="2019-04-24"
     DGX_SWBUILD_VERSION="4.1.0"
     DGX_COMMIT_ID="2c52306"
     DGX_PLATFORM="DGX Server for DGX-1"
     DGX_SERIAL_NUMBER="QTFCOU8220020"
     DISTRIB_ID=Ubuntu
     DISTRIB_RELEASE=18.04
     DISTRIB_CODENAME=bionic
     DISTRIB_DESCRIPTION="Ubuntu 18.04.2 LTS"
     NAME="Ubuntu"
     VERSION="18.04.2 LTS (Bionic Beaver)"
     ID=ubuntu
     ID_LIKE=debian
     PRETTY_NAME="Ubuntu 18.04.2 LTS"
     VERSION_ID="18.04"
     HOME_URL="https://www.ubuntu.com/"
     SUPPORT_URL="https://help.ubuntu.com/"
     BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
     PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
     VERSION_CODENAME=bionic
     UBUNTU_CODENAME=bionic
     Linux dgx13 4.15.0-47-generic #50-Ubuntu SMP Wed Mar 13 10:44:52 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
     
     ***GPU Information***
     Wed Jan 15 07:14:09 2020
     +-----------------------------------------------------------------------------+
     | NVIDIA-SMI 418.87.00    Driver Version: 418.87.00    CUDA Version: 10.1     |
     |-------------------------------+----------------------+----------------------+
     | GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
     | Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
     |===============================+======================+======================|
     |   0  Tesla V100-SXM2...  On   | 00000000:06:00.0 Off |                    0 |
     | N/A   34C    P0    55W / 300W |    518MiB / 32480MiB |      0%      Default |
     +-------------------------------+----------------------+----------------------+
     |   1  Tesla V100-SXM2...  On   | 00000000:07:00.0 Off |                    0 |
     | N/A   32C    P0    42W / 300W |     11MiB / 32480MiB |      0%      Default |
     +-------------------------------+----------------------+----------------------+
     |   2  Tesla V100-SXM2...  On   | 00000000:0A:00.0 Off |                    0 |
     | N/A   32C    P0    41W / 300W |     11MiB / 32480MiB |      0%      Default |
     +-------------------------------+----------------------+----------------------+
     |   3  Tesla V100-SXM2...  On   | 00000000:0B:00.0 Off |                    0 |
     | N/A   29C    P0    41W / 300W |     11MiB / 32480MiB |      0%      Default |
     +-------------------------------+----------------------+----------------------+
     |   4  Tesla V100-SXM2...  On   | 00000000:85:00.0 Off |                    0 |
     | N/A   33C    P0    42W / 300W |     11MiB / 32480MiB |      0%      Default |
     +-------------------------------+----------------------+----------------------+
     |   5  Tesla V100-SXM2...  On   | 00000000:86:00.0 Off |                    0 |
     | N/A   31C    P0    41W / 300W |     11MiB / 32480MiB |      0%      Default |
     +-------------------------------+----------------------+----------------------+
     |   6  Tesla V100-SXM2...  On   | 00000000:89:00.0 Off |                    0 |
     | N/A   35C    P0    42W / 300W |     11MiB / 32480MiB |      0%      Default |
     +-------------------------------+----------------------+----------------------+
     |   7  Tesla V100-SXM2...  On   | 00000000:8A:00.0 Off |                    0 |
     | N/A   31C    P0    40W / 300W |     11MiB / 32480MiB |      0%      Default |
     +-------------------------------+----------------------+----------------------+
     
     +-----------------------------------------------------------------------------+
     | Processes:                                                       GPU Memory |
     |  GPU       PID   Type   Process name                             Usage      |
     |=============================================================================|
     |    0     31029      C   ...ora/miniconda3/envs/cudf++12/bin/python   507MiB |
     +-----------------------------------------------------------------------------+
     
     ***CPU***
     Architecture:        x86_64
     CPU op-mode(s):      32-bit, 64-bit
     Byte Order:          Little Endian
     CPU(s):              80
     On-line CPU(s) list: 0-79
     Thread(s) per core:  2
     Core(s) per socket:  20
     Socket(s):           2
     NUMA node(s):        2
     Vendor ID:           GenuineIntel
     CPU family:          6
     Model:               79
     Model name:          Intel(R) Xeon(R) CPU E5-2698 v4 @ 2.20GHz
     Stepping:            1
     CPU MHz:             2504.974
     CPU max MHz:         3600.0000
     CPU min MHz:         1200.0000
     BogoMIPS:            4390.16
     Virtualization:      VT-x
     L1d cache:           32K
     L1i cache:           32K
     L2 cache:            256K
     L3 cache:            51200K
     NUMA node0 CPU(s):   0-19,40-59
     NUMA node1 CPU(s):   20-39,60-79
     Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cdp_l3 invpcid_single pti intel_ppin ssbd ibrs ibpb stibp tpr_shadow vnmi flexpriority ept vpid fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm cqm rdt_a rdseed adx smap intel_pt xsaveopt cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local dtherm ida arat pln pts flush_l1d
     
     ***CMake***
     /home/nfs/rzamora/miniconda3/envs/cudf++12/bin/cmake
     cmake version 3.16.2
     
     CMake suite maintained and supported by Kitware (kitware.com/cmake).
     
     ***g++***
     /usr/bin/g++
     g++ (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0
     Copyright (C) 2017 Free Software Foundation, Inc.
     This is free software; see the source for copying conditions.  There is NO
     warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
     
     
     ***nvcc***
     
     ***Python***
     /home/nfs/rzamora/miniconda3/envs/cudf++12/bin/python
     Python 3.7.6
     
     ***Environment Variables***
     PATH                            : /home/nfs/rzamora/.vscode-server/bin/26076a4de974ead31f97692a0d32f90d735645c0/bin:/home/nfs/rzamora/bin:/home/nfs/rzamora/.local/bin:/home/nfs/rzamora/miniconda3/envs/cudf++12/bin:/home/nfs/rzamora/miniconda3/condabin:/home/nfs/rzamora/.vscode-server/bin/26076a4de974ead31f97692a0d32f90d735645c0/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
     LD_LIBRARY_PATH                 :
     NUMBAPRO_NVVM                   :
     NUMBAPRO_LIBDEVICE              :
     CONDA_PREFIX                    : /home/nfs/rzamora/miniconda3/envs/cudf++12
     PYTHON_PATH                     :
     
     ***conda packages***
     /home/nfs/rzamora/miniconda3/condabin/conda
     # packages in environment at /home/nfs/rzamora/miniconda3/envs/cudf++12:
     #
     # Name                    Version                   Build  Channel
     _libgcc_mutex             0.1                 conda_forge    conda-forge
     aiohttp                   3.6.2                    pypi_0    pypi
     alabaster                 0.7.12                     py_0    conda-forge
     appdirs                   1.4.3                      py_1    conda-forge
     arrow-cpp                 0.15.0           py37h090bef1_2    conda-forge
     aspy.yaml                 1.3.0                      py_0    conda-forge
     async-timeout             3.0.1                    pypi_0    pypi
     attrs                     19.3.0                     py_0    conda-forge
     autoconf                  2.69            pl526h14c3975_9    conda-forge
     automake                  1.16.1               pl526_1003    conda-forge
     babel                     2.8.0                      py_0    conda-forge
     backcall                  0.1.0                      py_0    conda-forge
     black                     19.10b0                  py37_0    conda-forge
     bleach                    3.1.0                      py_0    conda-forge
     bokeh                     1.4.0                    py37_0    conda-forge
     boost-cpp                 1.70.0               h8e57a91_2    conda-forge
     brotli                    1.0.7             he1b5a44_1000    conda-forge
     bzip2                     1.0.8                h516909a_2    conda-forge
     c-ares                    1.15.0            h516909a_1001    conda-forge
     ca-certificates           2019.11.28           hecc5488_0    conda-forge
     cached-property           1.5.1                      py_0    conda-forge
     certifi                   2019.11.28               py37_0    conda-forge
     cffi                      1.13.2           py37h8022711_0    conda-forge
     cfgv                      2.0.1                      py_0    conda-forge
     chardet                   3.0.4                 py37_1003    conda-forge
     click                     7.0                        py_0    conda-forge
     cloudpickle               1.2.2                      py_1    conda-forge
     cmake                     3.16.2               h28c56e5_0    conda-forge
     cmake_setuptools          0.1.3                      py_0    rapidsai-nightly
     commonmark                0.9.1                      py_0    conda-forge
     cryptography              2.8              py37h72c5cf5_1    conda-forge
     cudatoolkit               10.1.243             h6bb024c_0    nvidia
     cudf                      0.12.0b0+852.g5e58253fe           dev_0    <develop>
     cudnn                     7.6.0                cuda10.1_0    nvidia
     cupy                      7.0.0                    pypi_0    pypi
     cython                    0.29.14          py37he1b5a44_0    conda-forge
     cytoolz                   0.10.1           py37h516909a_0    conda-forge
     dask                      2.9.1+7.g8530bdd0           dev_0    <develop>
     dask-core                 2.9.1                      py_0
     dask-cuda                 0.12.0a0+47.g90db937           dev_0    <develop>
     dask-cudf                 0.12.0b0+852.g5e58253fe           dev_0    <develop>
     dask-labextension         1.1.0                    pypi_0    pypi
     decorator                 4.4.1                      py_0    conda-forge
     defusedxml                0.6.0                      py_0    conda-forge
     distributed               2.9.1+6.gf68119a4           dev_0    <develop>
     dlpack                    0.2                  he1b5a44_1    conda-forge
     docutils                  0.15.2                   py37_0    conda-forge
     double-conversion         3.1.5                he1b5a44_2    conda-forge
     editdistance              0.5.3            py37he1b5a44_0    conda-forge
     entrypoints               0.3                   py37_1000    conda-forge
     expat                     2.2.5             he1b5a44_1004    conda-forge
     fastavro                  0.22.9           py37h516909a_0    conda-forge
     fastparquet               0.3.2                    pypi_0    pypi
     fastrlock                 0.4             py37he1b5a44_1000    conda-forge
     flake8                    3.7.9                    py37_0    conda-forge
     flatbuffers               1.11.0               he1b5a44_0    conda-forge
     freetype                  2.10.0               he983fc9_1    conda-forge
     fsspec                    0.6.2                      py_0    conda-forge
     future                    0.18.2                   py37_0    conda-forge
     gflags                    2.2.2             he1b5a44_1002    conda-forge
     glog                      0.4.0                he1b5a44_1    conda-forge
     gmp                       6.1.2             hf484d3e_1000    conda-forge
     grpc-cpp                  1.23.0               h18db393_0    conda-forge
     heapdict                  1.0.1                      py_0    conda-forge
     hypothesis                5.1.1                      py_0    conda-forge
     icu                       64.2                 he1b5a44_1    conda-forge
     identify                  1.4.9                      py_0    conda-forge
     idna                      2.8                   py37_1000    conda-forge
     imagesize                 1.2.0                      py_0    conda-forge
     importlib_metadata        1.3.0                    py37_0    conda-forge
     ipykernel                 5.1.3            py37h5ca1d4c_0    conda-forge
     ipython                   7.11.1           py37h5ca1d4c_0    conda-forge
     ipython_genutils          0.2.0                      py_1    conda-forge
     isort                     4.3.21                   py37_0    conda-forge
     jedi                      0.15.2                   py37_0    conda-forge
     jeepney                   0.4.2                    pypi_0    pypi
     jinja2                    2.10.3                     py_0    conda-forge
     jpeg                      9c                h14c3975_1001    conda-forge
     json5                     0.8.5                      py_0
     jsonschema                3.2.0                    py37_0    conda-forge
     jupyter-server-proxy      1.2.0                    pypi_0    pypi
     jupyter_client            5.3.4                    py37_0    conda-forge
     jupyter_core              4.6.1                    py37_0    conda-forge
     jupyterlab                1.2.4              pyhf63ae98_0
     jupyterlab-nvdashboard    0.1.11                   pypi_0    pypi
     jupyterlab_server         1.0.6                      py_0
     keyring                   21.0.0                   pypi_0    pypi
     krb5                      1.16.4               h2fd8d38_0    conda-forge
     ld_impl_linux-64          2.33.1               h53a641e_7    conda-forge
     libblas                   3.8.0               14_openblas    conda-forge
     libcblas                  3.8.0               14_openblas    conda-forge
     libcurl                   7.65.3               hda55be3_0    conda-forge
     libedit                   3.1.20170329      hf8c457e_1001    conda-forge
     libevent                  2.1.10               h72c5cf5_0    conda-forge
     libffi                    3.2.1             he1b5a44_1006    conda-forge
     libgcc-ng                 9.2.0                h24d8f2e_1    conda-forge
     libgfortran-ng            7.3.0                hdf63c60_3    conda-forge
     libgomp                   9.2.0                h24d8f2e_1    conda-forge
     libhwloc                  2.1.0                h3c4fd83_0    conda-forge
     libiconv                  1.15              h516909a_1005    conda-forge
     liblapack                 3.8.0               14_openblas    conda-forge
     libllvm8                  8.0.1                hc9558a2_0    conda-forge
     libopenblas               0.3.7                h5ec1e0e_6    conda-forge
     libpng                    1.6.37               hed695b0_0    conda-forge
     libprotobuf               3.8.0                h8b12597_0    conda-forge
     librmm                    0.12.0a200108      cuda10.1_153    rapidsai-nightly
     libsodium                 1.0.17               h516909a_0    conda-forge
     libssh2                   1.8.2                h22169c7_2    conda-forge
     libstdcxx-ng              9.2.0                hdf63c60_1    conda-forge
     libtiff                   4.1.0                hfc65ed5_0    conda-forge
     libtool                   2.4.6             h14c3975_1002    conda-forge
     libuv                     1.34.0               h516909a_0    conda-forge
     libxml2                   2.9.10               hee79883_0    conda-forge
     llvmlite                  0.30.0           py37h8b12597_1    conda-forge
     locket                    0.2.0                      py_2    conda-forge
     lz4-c                     1.8.3             he1b5a44_1001    conda-forge
     m4                        1.4.18            h14c3975_1001    conda-forge
     make                      4.2.1             h14c3975_2004    conda-forge
     markdown                  3.0.1                    pypi_0    pypi
     markupsafe                1.1.1            py37h516909a_0    conda-forge
     mccabe                    0.6.1                      py_1    conda-forge
     mistune                   0.8.4           py37h516909a_1000    conda-forge
     more-itertools            8.0.2                      py_0    conda-forge
     msgpack-python            0.6.2            py37hc9558a2_0    conda-forge
     multidict                 4.7.3                    pypi_0    pypi
     mypy_extensions           0.4.3                    py37_0    conda-forge
     nbconvert                 5.6.1                    py37_0    conda-forge
     nbformat                  5.0.3                      py_0    conda-forge
     nbsphinx                  0.5.0                      py_0    conda-forge
     nccl                      2.4.6.1              cuda10.1_0    nvidia
     ncurses                   6.1               hf484d3e_1002    conda-forge
     nodeenv                   1.3.4                      py_0    conda-forge
     nodejs                    10.13.0              he6710b0_0
     notebook                  6.0.1                    py37_0    conda-forge
     numba                     0.46.0           py37hb3f55d8_1    conda-forge
     numpy                     1.17.3           py37h95a1406_0    conda-forge
     numpydoc                  0.9.2                      py_0    conda-forge
     nvstrings-cudaunknown     0.0.0.dev0               pypi_0    pypi
     olefile                   0.46                       py_0    conda-forge
     openmp_impl               4.5                       0_gnu    conda-forge
     openssl                   1.1.1d               h516909a_0    conda-forge
     packaging                 20.0                       py_0    conda-forge
     pandas                    0.25.3           py37he6710b0_0
     pandoc                    1.19.2                        0    conda-forge
     pandocfilters             1.4.2                      py_1    conda-forge
     parquet-cpp               1.5.1                         2    conda-forge
     parso                     0.5.2                      py_0    conda-forge
     partd                     1.1.0                      py_0    conda-forge
     pathspec                  0.7.0                      py_0    conda-forge
     perl                      5.26.2            h516909a_1006    conda-forge
     pexpect                   4.7.0                    py37_0    conda-forge
     pickleshare               0.7.5                 py37_1000    conda-forge
     pillow                    7.0.0            py37hefe7db6_0    conda-forge
     pip                       19.3.1                   py37_0    conda-forge
     pkg-config                0.29.2            h516909a_1006    conda-forge
     pkginfo                   1.5.0.1                  pypi_0    pypi
     pluggy                    0.13.0                   py37_0    conda-forge
     pre_commit                1.18.1                   py37_0    conda-forge
     prometheus_client         0.7.1                      py_0    conda-forge
     prompt_toolkit            3.0.2                      py_0    conda-forge
     psutil                    5.6.7            py37h516909a_0    conda-forge
     ptyprocess                0.6.0                   py_1001    conda-forge
     py                        1.8.1                      py_0    conda-forge
     pyarrow                   0.15.0           py37h8b68381_1    conda-forge
     pycodestyle               2.5.0                      py_0    conda-forge
     pycparser                 2.19                     py37_1    conda-forge
     pyflakes                  2.1.1                      py_0    conda-forge
     pygments                  2.5.2                      py_0    conda-forge
     pynvml                    8.0.4                     dev_0    <develop>
     pyopenssl                 19.1.0                   py37_0    conda-forge
     pyparsing                 2.4.6                      py_0    conda-forge
     pyrsistent                0.15.7           py37h516909a_0    conda-forge
     pysocks                   1.7.1                    py37_0    conda-forge
     pytest                    5.3.2                    py37_0    conda-forge
     python                    3.7.6                h357f687_2    conda-forge
     python-dateutil           2.8.1                      py_0    conda-forge
     python-snappy             0.5.4            py37hee44bf9_1    conda-forge
     pytz                      2019.3                     py_0    conda-forge
     pyyaml                    5.3              py37h516909a_0    conda-forge
     pyzmq                     18.1.1           py37h1768529_0    conda-forge
     rapidjson                 1.1.0             he1b5a44_1002    conda-forge
     re2                       2020.01.01           he1b5a44_0    conda-forge
     readline                  8.0                  hf8c457e_0    conda-forge
     readme-renderer           24.0                     pypi_0    pypi
     recommonmark              0.6.0                      py_0    conda-forge
     regex                     2020.1.8         py37h516909a_0    conda-forge
     requests                  2.22.0                   py37_1    conda-forge
     requests-toolbelt         0.9.1                    pypi_0    pypi
     rhash                     1.3.6             h14c3975_1001    conda-forge
     rmm                       0.12.0a200108          py37_153    rapidsai-nightly
     secretstorage             3.1.2                    pypi_0    pypi
     send2trash                1.5.0                      py_0    conda-forge
     setuptools                44.0.0                   py37_0    conda-forge
     simpervisor               0.3                      pypi_0    pypi
     six                       1.13.0                   py37_0    conda-forge
     snappy                    1.1.7             he1b5a44_1003    conda-forge
     snowballstemmer           2.0.0                      py_0    conda-forge
     sortedcontainers          2.1.0                      py_0    conda-forge
     sphinx                    2.3.1                      py_0    conda-forge
     sphinx-markdown-tables    0.0.10                   pypi_0    pypi
     sphinx_rtd_theme          0.4.3                      py_0    conda-forge
     sphinxcontrib-applehelp   1.0.1                      py_0    conda-forge
     sphinxcontrib-devhelp     1.0.1                      py_0    conda-forge
     sphinxcontrib-htmlhelp    1.0.2                      py_0    conda-forge
     sphinxcontrib-jsmath      1.0.1                      py_0    conda-forge
     sphinxcontrib-qthelp      1.0.2                      py_0    conda-forge
     sphinxcontrib-serializinghtml 1.1.3                      py_0    conda-forge
     sphinxcontrib-websupport  1.1.2                      py_0    conda-forge
     sqlite                    3.30.1               hcee41ef_0    conda-forge
     streamz                   0.5.2                    pypi_0    pypi
     tblib                     1.6.0                      py_0    conda-forge
     terminado                 0.8.3                    py37_0    conda-forge
     testpath                  0.4.4                      py_0    conda-forge
     thrift                    0.13.0                   pypi_0    pypi
     thrift-cpp                0.12.0            hf3afdfd_1004    conda-forge
     tk                        8.6.10               hed695b0_0    conda-forge
     toml                      0.10.0                     py_0    conda-forge
     toolz                     0.10.0                     py_0    conda-forge
     torch                     1.3.1                    pypi_0    pypi
     tornado                   6.0.3            py37h516909a_0    conda-forge
     tqdm                      4.41.1                   pypi_0    pypi
     traitlets                 4.3.3                    py37_0    conda-forge
     twine                     3.1.1                    pypi_0    pypi
     typed-ast                 1.4.0            py37h516909a_0    conda-forge
     typing_extensions         3.7.4.1                  py37_0    conda-forge
     ucxpy                     0.12.0a0+83.g553a09e           dev_0    <develop>
     uriparser                 0.9.3                he1b5a44_1    conda-forge
     urllib3                   1.25.7                   py37_0    conda-forge
     virtualenv                16.7.5                     py_0    conda-forge
     wcwidth                   0.1.8                      py_0    conda-forge
     webencodings              0.5.1                      py_1    conda-forge
     wheel                     0.33.6                   py37_0    conda-forge
     xz                        5.2.4             h14c3975_1001    conda-forge
     yaml                      0.2.2                h516909a_1    conda-forge
     yarl                      1.4.2                    pypi_0    pypi
     zeromq                    4.3.2                he1b5a44_2    conda-forge
     zict                      1.0.0                      py_0    conda-forge
     zipp                      0.6.0                      py_0    conda-forge
     zlib                      1.2.11            h516909a_1006    conda-forge
     zstd                      1.4.3                h3b9ef0a_0    conda-forge
     
</pre></details>

**Additional context**
Relevent to [recsys#11](https://github.com/rapidsai/recsys/pull/11) - cc @jperez999

**Comment by shwina** [comment]
Definitely weird. Have a feeling this has something to do with the `legacy/replace.cu` code. Looking..

**Comment by shwina** [comment]
So I *think* that this is an issue in the underlying (legacy) libcudf `replace` function. If that's the case, a hacky way to fix this from the Python side would be:

```python
class Series:
        def _prune_mask(self):
            if self._column.nullable and self.null_count == 0:
                self._column.mask = None
```

```python
import cudf
import numpy as np

_cats = ["cat", "dog", "bird"]
series = cudf.Series(np.random.choice(_cats, 100))
encoded = series.label_encoding(_cats)
encoded._prune_mask()
encoded.replace(0, 3)
```

A less disruptive solution would be to implement `_prune_mask()` in `Column`, and perform the prune in `Column.replace()`.

**Comment by rjzamora** [comment]
Thanks @shwina!  Just curious - Is there a libcudf++ version of replace in the works?

**Comment by shwina** [comment]
Yup - there's definitely a libcudf++ `replace` and we'll plumb it in, but very likely not for 0.12

**Comment by kkraus14** [comment]
Will be fixed by libcudf++ refactor so defer to 0.13 for it.

**Comment by rgsl888prabhu** [comment]
Think this has been resolved due to refactoring and can be closed.
